src/users/usersActions.js

```diff
@@ -25,17 +26,41 @@ export const reportPresence = (isActive: boolean = true, newUserInput: boolean =
   });
 };
 
-const typingWorker = auth => ({
-  get_current_time: () => new Date().getTime(),
+const typingWorker = (state: GlobalState) => {
+  const auth: Auth = getAuth(state);
 
-  notify_server_start: (user_ids_array: number[]) => {
-    api.typing(auth, JSON.stringify(user_ids_array), 'start');
-  },
+  const getRecipients = user_ids_array => {
+    // User ID arrays are only supported in server versions >= 2.0.0-rc1, for
+    // versions before this, email arrays are used. If current server version is
+    // undetermined, user ID arrays are optimistically used.
+    let useEmailArrays: boolean;
+    const activeAccount: Account | void = tryGetActiveAccount(state);
 
-  notify_server_stop: (user_ids_array: number[]) => {
-    api.typing(auth, JSON.stringify(user_ids_array), 'stop');
-  },
-});
+    if (!activeAccount || activeAccount.zulipVersion === undefined) {
+      useEmailArrays = false;
+    } else {
+      useEmailArrays = !new ZulipVersion(activeAccount.zulipVersion).isAtLeast(
+        new ZulipVersion('2.0.0-rc1'),
```

Comment:
This could use a comment referencing the actual server commit in which this capability was added.

Reply:
That's 2f634f8c06eaece269b9a2692c8ec8edc5af6ac5

Reply:
(For later readers' reference, this is `1.9.0-1173-g2f634f8c06`, or equivalently, `2.0.0-rc1~476`.)